test/coordinate/cartesian_differential.cpp

```diff
@@ -0,0 +1,252 @@
+    BOOST_TEST((std::is_same<decltype(motion2.get_dy()),
+        quantity<bu::divide_typeof_helper<si::length, si::time>::type>>::value));
+    BOOST_TEST((std::is_same<decltype(motion2.get_dz()),
+        quantity<bu::divide_typeof_helper<si::length, si::time>::type>>::value));
+
+    //constructing from spherical_coslat differential
+    auto spherical_coslat_motion = make_spherical_coslat_differential
+    (0.523599 * si::radian, 60.0 * bud::degrees, 1.0 * meters/seconds);
+    auto motion3 = make_cartesian_differential(spherical_coslat_motion);
+    BOOST_CHECK_CLOSE(motion3.get_dx().value(), 0.8100222, 0.001);
+    BOOST_CHECK_CLOSE(motion3.get_dy().value(), 0.467666778, 0.001);
+    BOOST_CHECK_CLOSE(motion3.get_dz().value(), 0.353768031, 0.001);
+
+    //checking whether quantity stored is as expected or not
+    BOOST_TEST((std::is_same<decltype(motion3.get_dx()),
+        quantity<bu::divide_typeof_helper<si::length, si::time>::type>>::value));
+    BOOST_TEST((std::is_same<decltype(motion3.get_dy()),
+        quantity<bu::divide_typeof_helper<si::length, si::time>::type>>::value));
+    BOOST_TEST((std::is_same<decltype(motion3.get_dz()),
+        quantity<bu::divide_typeof_helper<si::length, si::time>::type>>::value));
+}
+
+BOOST_AUTO_TEST_SUITE_END()
+
+BOOST_AUTO_TEST_SUITE(cartesian_differential_operators)
+
+BOOST_AUTO_TEST_CASE(addition)
+{
+    auto motion1 = make_cartesian_differential
+        (11.0*meters/seconds, 15.0*meters/seconds, 19.0*meters/seconds);
+    auto motion2 = make_cartesian_differential
+        (6.0*meters/seconds, 10.0*meters/seconds, 11.0*meters/seconds);
+
+    auto sum = make_cartesian_differential(motion1 + motion2);
+
+    BOOST_CHECK_CLOSE(sum.get_dx().value(), 17.0, 0.001);
+    BOOST_CHECK_CLOSE(sum.get_dy().value(), 25, 0.001);
+    BOOST_CHECK_CLOSE(sum.get_dz().value(), 30, 0.001);
+
+    //checking whether quantity stored is as expected or not
+    BOOST_TEST((std::is_same<decltype(sum.get_dx()),
+        quantity<bu::divide_typeof_helper<si::length, si::time>::type>>::value));
+    BOOST_TEST((std::is_same<decltype(sum.get_dy()),
+        quantity<bu::divide_typeof_helper<si::length, si::time>::type>>::value));
+    BOOST_TEST((std::is_same<decltype(sum.get_dz()),
+        quantity<bu::divide_typeof_helper<si::length, si::time>::type>>::value));
+}
+
+BOOST_AUTO_TEST_CASE(multiplication)
+{
+    auto motion1 = make_cartesian_differential
+        (3.0*meters/seconds, 9.0*meters/seconds, 6.0*meters/seconds);
+
+    auto product = make_cartesian_differential
+        (motion1 * quantity<si::time>(5*seconds));
+
+    BOOST_CHECK_CLOSE(product.get_dx().value(), 15.0, 0.001);
+    BOOST_CHECK_CLOSE(product.get_dy().value(), 45.0, 0.001);
+    BOOST_CHECK_CLOSE(product.get_dz().value(), 30.0, 0.001);
+
+    //checking whether quantity stored is as expected or not
+    BOOST_TEST((std::is_same<decltype(product.get_dx()),
```

Comment:
if it is velocity is multiplied with time then isn't it supposed to be length only?